From 6cffd344d0a89dc2ee6dc24ec1f3ec5e5efe984d Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hmehrtens@maxlinear.com>
Date: Fri, 12 Feb 2021 17:42:20 +0100
Subject: [PATCH] PONRTSYS-7990: Allow access to the CGU and RCU in NGI
 firewall

Do not block access from the InterAptiv to the RCU and CGU registers in
the NGI firewall any more. With these changes the ICC regmap  driver should
not be needed any more.
---
 arch/mips/lantiq/grx500_bootcore/ngi_prx300.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/mips/lantiq/grx500_bootcore/ngi_prx300.c b/arch/mips/lantiq/grx500_bootcore/ngi_prx300.c
index 070608439a19..4b8787f7dd6e 100644
--- a/arch/mips/lantiq/grx500_bootcore/ngi_prx300.c
+++ b/arch/mips/lantiq/grx500_bootcore/ngi_prx300.c
@@ -277,13 +277,13 @@ void set_permission(void)
 	permission_wr(ssx4_base,
 			REG4_TREG4_PM_TREG4_PM_READ_PERMISSION_0, M4KEC_ONLY);
 	permission_wr(ssx4_base,
-			REG4_TCGU_PM_TCGU_PM_WRITE_PERMISSION_0, M4KEC_ONLY | CPU3);
+			REG4_TCGU_PM_TCGU_PM_WRITE_PERMISSION_0, IAP_N_4KEC);
 	permission_wr(ssx4_base,
-			REG4_TCGU_PM_TCGU_PM_READ_PERMISSION_0, M4KEC_ONLY | CPU3);
+			REG4_TCGU_PM_TCGU_PM_READ_PERMISSION_0, IAP_N_4KEC);
 	permission_wr(ssx4_base,
-			REG4_TRCU_PM_TRCU_PM_WRITE_PERMISSION_0, M4KEC_ONLY);
+			REG4_TRCU_PM_TRCU_PM_WRITE_PERMISSION_0, IAP_N_4KEC);
 	permission_wr(ssx4_base,
-			REG4_TRCU_PM_TRCU_PM_READ_PERMISSION_0, M4KEC_ONLY);
+			REG4_TRCU_PM_TRCU_PM_READ_PERMISSION_0, IAP_N_4KEC);
 	permission_wr(ssx4_base,
 			REG4_TPMU_PM_TPMU_PM_WRITE_PERMISSION_0, IAP_N_4KEC);
 	permission_wr(ssx4_base,
